hathor/websocket/streamer.py

```diff
@@ -72,23 +94,34 @@ def __init__(self,
 
         self.deferred: Deferred[bool] = Deferred()
 
-        # Statistics.
+        # Statistics
+        # ----------
         self.stats_log_interval = self.STATS_LOG_INTERVAL
         self.stats_total_messages: int = 0
         self.stats_sent_addresses: int = 0
         self.stats_sent_vertices: int = 0
 
-        # Execution control.
-        self._started = False
-        self._is_running = False
-        self._paused = False
-        self._stop = False
+        # Execution control
+        # -----------------
+        self._state = StreamerState.NOT_STARTED
+        # Used to mark that the streamer is currently running its main loop and sending messages.
+        self._is_main_loop_running = False
+        # Used to mark that the streamer was paused by the transport layer.
+        self._is_paused_by_transport = False
```

Comment:
What's the difference between those and the `ACTIVE` and `PAUSED` states?

Reply:
You're right about the `PAUSED` state. I just refactored out the `self._is_paused_by_transport`. Nice catch! Thanks!

I answered about the `ACTIVE` in this thread: https://github.com/HathorNetwork/hathor-core/pull/1129#discussion_r1747376634